rackcity/views/decommission_views.py

```diff
@@ -66,23 +68,38 @@ def decommission_asset_parameterized(data, query_params, user):
         response = get_cp_already_executed_response(change_plan)
         if response:
             return response
-        if (
-            not AssetCP.objects.filter(id=id).exists()
-            and Asset.objects.filter(id=id).exists()
-        ):
-            existing_asset = Asset.objects.get(id=id)
+        assets, assets_cp = get_assets_for_cp(change_plan.id, show_decommissioned=True)
+        if assets_cp.filter(related_asset=id).exists():
+            decommissioned_asset_cp = assets_cp.get(related_asset=id)
+            decommissioned_asset_cp.is_decommissioned = True
+        elif assets_cp.filter(id=id).exists():
+            decommissioned_asset_cp = assets_cp.get(id=id)
+            decommissioned_asset_cp.is_decommissioned = True
+        elif assets.filter(id=id).exists():
+            existing_asset = assets.get(id=id)
```

Comment:
updating decommissioning on change plan -> because of new routing if an asset is modified on change plan, the front end keeps track of the id of its related asset.